### PR TITLE
Fix camera crash, take X

### PR DESCRIPTION
**Pull request by nirvn**
Removes legacy encryption from openssl, see https://www.openssl.org/docs/man3.0/man7/OSSL_PROVIDER-legacy.html

### CODE REVIEW

File: vcpkg/overlay/qgis-qt6/portfile.cmake [hunk 1]
```diff
@@ -46,6 +46,7 @@ list(APPEND QGIS_OPTIONS "-DFLEX_EXECUTABLE=${FLEX}")
 list(APPEND QGIS_OPTIONS "-DQGIS_INCLUDE_SUBDIR=include/qgis")
 list(APPEND QGIS_OPTIONS "-DBUILD_WITH_QT6=ON")
 list(APPEND QGIS_OPTIONS "-DQGIS_MACAPP_FRAMEWORK=FALSE")
+list(APPEND QGIS_OPTIONS "-DCMAKE_DISABLE_FIND_PACKAGE_HDF5=TRUE")
```
Comment (nirvn) [inline vcpkg/overlay/qgis-qt6/portfile.cmake:49]:
```suggestion
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c65f9288c4c610cdc1b72c4e0a81481e95534dcb: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4898-579dd94ac5b175029ae3718ece2b41481434bff6-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4898-579dd94ac5b175029ae3718ece2b41481434bff6-arm-neon-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4898-579dd94ac5b175029ae3718ece2b41481434bff6-x64-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4898-579dd94ac5b175029ae3718ece2b41481434bff6-arm64-android.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4898-579dd94ac5b175029ae3718ece2b41481434bff6-arm-neon-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4898-579dd94ac5b175029ae3718ece2b41481434bff6-x64-android.apk)
</details>

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -5,14 +5,15 @@ on:
     branches:
       - master
       - release-**
+      - android_more_dynamic
```
Comment (m-kuhn) [inline .github/workflows/android.yml:8]:
```suggestion
 ```

File: src/app/CMakeLists.txt [hunk 3]
```diff
@@ -72,13 +72,11 @@ function(create_executable)
...
 
       set(PROJECT_NAME ${exe_TARGET})
-      include(FetchContent)
-      FetchContent_Declare(
-        android_openssl
-        GIT_REPOSITORY https://github.com/KDAB/android_openssl.git
-        GIT_TAG ef412c6ebf131fae29a873d0c5db6c6b9dd494fd)
-      FetchContent_MakeAvailable(android_openssl)
-      include(${FETCHCONTENT_BASE_DIR}/android_openssl-src/CMakeLists.txt)
+      # include(FetchContent) FetchContent_Declare( android_openssl
```
Comment (m-kuhn) [inline src/app/CMakeLists.txt:75]: Todo: delete

File: src/app/main.cpp [hunk 4]
```diff
@@ -52,7 +52,7 @@
 
 #if HAVE_STATIC_QCA_PLUGINS
 #include <QtPlugin>
-Q_IMPORT_PLUGIN( opensslPlugin )
+// Q_IMPORT_PLUGIN( opensslPlugin )
```
Comment (m-kuhn) [inline src/app/main.cpp:55]: Review

Comment (nirvn) [reply src/app/main.cpp:55]: @m-kuhn , cause of crash:

--------- beginning of crash
12-31 23:03:56.555 18204 18239 E AndroidRuntime: FATAL EXCEPTION: qtMainLoopThread
12-31 23:03:56.555 18204 18239 E AndroidRuntime: Process: ch.opengis.qfield_beta, PID: 18204
12-31 23:03:56.555 18204 18239 E AndroidRuntime: java.lang.UnsatisfiedLinkError: dlopen failed: library "libqca-ossl.so" not found: needed by /data/app/~~PXkdAMRHMsvMmkmuj4n_sg==/ch.opengis.qfield_beta-Nuz0-nhYGdz96d8L236v8A==/lib/arm64/libqfield_arm64-v8a.so in namespace clns-4

Comment (nirvn) [reply src/app/main.cpp:55]: The .so is indeed missing from the APK.

File: .github/workflows/android.yml [hunk 5]
```diff
@@ -107,39 +100,25 @@ jobs:
...
-        run: |
-          echo "ndk.dir=$ANDROID_NDK_HOME" >> local.properties
-          echo "yes" | sudo /usr/local/lib/android/sdk/tools/bin/sdkmanager --uninstall "ndk-bundle"
-          echo "yes" | sudo /usr/local/lib/android/sdk/tools/bin/sdkmanager "platforms;android-33" "build-tools;$ANDROID_BUILD_TOOLS_VERSION" "ndk;$ANDROID_NDK_VERSION" tools platform-tools
-          echo "yes" | sudo /usr/local/lib/android/sdk/tools/bin/sdkmanager --licenses
-
       - name: 🌱 Install dependencies and generate project files
         env:
           WORKSPACE: ${{ github.workspace }}
+          ANDROID_SDK_ROOT: /usr/local/lib/android/sdk/
```
Comment (m-kuhn) [inline .github/workflows/android.yml:106]: Preset by github, remove

**Review by m-kuhn — APPROVED** [review]

**Comment by qfield-fairy** [comment]
The backport to `release-3_1` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply 28a57f0c5... Working local docker Android builds with self-built Qt
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
[backport-4898-to-release-3_1 59abb4e8d] Build Qt for android
 Author: Matthias Kuhn <matthias@opengis.ch>
 Date: Sat Dec 23 17:19:49 2023 +0100
 6 files changed, 9 insertions(+), 23 deletions(-)
[backport-4898-to-release-3_1 2b2430a1a] Add qtbase overlay
 Author: Matthias Kuhn <matthias@opengis.ch>
 Date: Sat Dec 23 17:21:29 2023 +0100
 24 files changed, 1960 insertions(+)
 create mode 100644 vcpkg/overlay/qtbase/GLIB2-static.patch
 create mode 100644 vcpkg/overlay/qtbase/allow_outside_prefix.patch
 create mode 100644 vcpkg/overlay/qtbase/clang-cl_QGADGET_fix.diff
 create mode 100644 vcpkg/overlay/qtbase/clang-cl_source_location.patch
 create mode 100644 vcpkg/overlay/qtbase/cmake/qt_install_copyright.cmake
 create mode 100644 vcpkg/overlay/qtbase/cmake/qt_install_submodule.cmake
 create mode 100644 vcpkg/overlay/qtbase/cmake/qt_port_data.cmake
 create mode 100644 vcpkg/overlay/qtbase/cmake/qt_port_details.cmake
 create mode 100644 vcpkg/overlay/qtbase/config_install.patch
 create mode 100644 vcpkg/overlay/qtbase/dont_force_cmakecache_latest.patch
 create mode 100644 vcpkg/overlay/qtbase/env.patch
 create mode 100644 vcpkg/overlay/qtbase/fix-host-aliasing.patch
 create mode 100644 vcpkg/overlay/qtbase/fix_cmake_build.patch
 create mode 100644 vcpkg/overlay/qtbase/fix_cmake_build_type.patch
 create mode 100644 vcpkg/overlay/qtbase/fix_egl.patch
 create mode 100644 vcpkg/overlay/qtbase/fix_egl_2.patch
 create mode 100644 vcpkg/overlay/qtbase/harfbuzz.patch
 create mode 100644 vcpkg/overlay/qtbase/installed_dir.patch
 create mode 100644 vcpkg/overlay/qtbase/portfile.cmake
 create mode 100644 vcpkg/overlay/qtbase/qmake.debug.bat
 create mode 100644 vcpkg/overlay/qtbase/qt.conf.in
 create mode 100644 vcpkg/overlay/qtbase/qtpaths.debug.bat
 create mode 100644 vcpkg/overlay/qtbase/vcpkg.json
 create mode 100644 vcpkg/overlay/qtbase/windeployqt.debug.bat
[backport-4898-to-release-3_1 06354cd34] Enable egl support for qtbase/android
 Author: Matthias Kuhn <matthias@opengis.ch>
 Date: Sat Dec 23 17:21:57 2023 +0100
 1 file changed, 1 insertion(+), 1 deletion(-)
[backport-4898-to-release-3_1 bc09d357b] Install required deps
 Author: Matthias Kuhn <matthias@opengis.ch>
 Date: Sat Dec 23 17:26:44 2023 +0100
 2 files changed, 6 insertions(+)
[backport-4898-to-release-3_1 0be9556cf] Diable default feature egl for linux
 Author: Matthias Kuhn <matthias@opengis.ch>
 Date: Sat Dec 23 18:00:33 2023 +0100
 1 file changed, 1 deletion(-)
[backport-4898-to-release-3_1 b6b334a91] We only need qttools on host
 Author: Matthias Kuhn <matthias@opengis.ch>
 Date: Sat Dec 23 21:07:22 2023 +0100
 2 files changed, 8 insertions(+), 2 deletions(-)
Auto-merging cmake/Platform.cmake
CONFLICT (content): Merge conflict in cmake/Platform.cmake

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-release-3_1 release-3_1
# Navigate to the new working tree
cd .worktrees/backport-release-3_1
# Create a new branch
git switch --create backport-4898-to-release-3_1
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 44f608a0b9ba0e8cba473c2b72a67b3fe1ab874c,f62237445b0c9b61688c7c3536f44e0fc870f78f,5675637a5c1ca33e651564f069b23a49fe0072ab,c6177858d928695199a25c069f3bd92a0eb40360,6bcced59574b55a62d3a0bf48de23e4fc04b1303,728d7a307f3d2caf6645335e2d89bfea3b77598c,28a57f0c5139450c738a3028cd70eccd2dc03ddf,3eb353d106e6338fc229064ec5d81690dd347ff7,f1527bf6db472e15176041700933fbdc678dd991,c3594429ae61ef0de12d433aa2782ff66ffa7b13,bd16fe5b0e184774bfb5b8aa5ed193a678b63cb2,44f63d97d64a16752d88a75817f65a0087dd96fb,6659fd14c0be38761d259625fe992b2e7798f2ab,7391cc979d03a7f44836331045e55ffaa42b9fa7,294377247d9af98ec25ffdd40eb693905352bbbf,c89ebe2ac6d3553d41f7d5012691de3c6e81dba4,d57cedb84b952f63e4c5c2f42b162e3fe1ce6490,1e80200fdd966407c9575f17c2f61884b449053d,51262e43979c6d87f62c49502302db6fde160291,3ea31f2ff2049e37101b2c3132427ff73c472a2d,ac1ce74e43aa168fb3cfb235e2b456a7d82ffd85,373441081acd4bcc1cb48aaaf07fc2dfb9da864c,23195f8a6e8245e28e2fa0c21846e7e8de60d0f0,604b19fa1eaf18e7ba022fb3d39dabc3232b827a,a0746ed14a82ced498628435f08abee261e8f3dd,a7283f102c3522abcc56a1015ce7231ad5d90736,706bd06d1e391264e452e6b1b3e2398f42a5b50f,4511edad269a5e56b6732bc2f726037d1b47522f,e19ba5cb4522b97acbe355be98c833ec2d1d7ad5,688b89f2a789b0e67b13eed1d688338313ffae03
# Push it to GitHub
git push --set-upstream origin backport-4898-to-release-3_1
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-release-3_1
```
Then, create a pull request where the `base` branch is `release-3_1` and the `compare`/`head` branch is `backport-4898-to-release-3_1`.

**Pull request merged**